### DISCUSSION TITLE
Release 0.6.8 (1.21.20.22)

### DISCUSSION THREAD

**Discussion by m-brisch** [Mojang/minecraft-editor#184] [Announcements]
_[Read the previous release notes here](https://github.com/Mojang/minecraft-editor/discussions/181)_  

---    

## Summary  

Hello and happy preview day 🥳! We are now on 0.6.8 (1.21.20.22), as of Wednesday, July 10th, 2024. 

As a reminder of Editor API support:   

- Editor Extensions [overview article](https://learn.microsoft.com/en-us/minecraft/creator/documents/editorextensionsintroduction) and [API reference documentation](https://aka.ms/EditorAPI) on learn.microsoft.com  
- [Starter Kit](https://github.com/Mojang/minecraft-editor-extension-starter-kit): Contains the compiler tools and various libraries required to get started making Editor Extensions  
- [Samples](https://github.com/Mojang/minecraft-editor-extension-samples) created by our team to jump-start your own Extension creations  

Additionally, if you're new to the Editor GitHub this week, we encourage you to join the Bedrock Add-Ons Discord [#Editor channel](https://discord.gg/TZnQKFwGpt)!  

---

## Updated Feature: Deferred Lighting panel
We added a few quality-of-life improvements to the Deferred Lighting panel. Numeric inputs have been replaced with sliders (which now support non-integer values) to more easily configure your settings. We also addressed a bug that would producing visual artifacts within Editor so you should no longer have to change to crosshair mode to accurately view the world lighting.

For more information about Deferred Lighting, check out the [learn page](https://learn.microsoft.com/en-us/minecraft/creator/documents/deferredlighting?view=minecraft-bedrock-stable).

---

## New Feature: Entity Transactions
We updated the summon tool to use `TransactionManager` to track the addition and deletion of entities. This means you're now able to use undo/redo to more easily create entity rich environments.

---

## New Feature: Attach Debugger
To make it easier to debug your own Editor Extensions we added a button you can add to your Action Bar which will attempt to connect to the Minecraft script debugger on port 19144.

For more information about the Minecraft script debugger, check out the [GitHub page](https://github.com/Mojang/minecraft-debugger)

---

## Change Log   
- Updated set of blocks + entities that appear within the Editor picker
- Added 'Attach Debugger' ActionBarItem shortcut for attaching to the scripting debugger 
- Implemented single entity transaction payload for entity delete/create 
- Implemented Undo/Redo of single entity creation and deletion
- Added isFloat to support floating point values in addNumber Editor Property Pane API 
- Added copy to clipboard button to log panel messages
- Added localized field names into the Deferred Lighting property editor dropdown
- Added Global Illumination settings to the Deferred Lighting property editor
- Fixed bug that caused visual artifacts when enabling Deferred Lighting in Editor
- Removed fallback display text properties from API in favor of single localized property. 
  - Added labelproperty to IDropdownItem API, removed displayAltText and displayStringId properties
  - Added labelproperty to IMenuCreationParams API, removed name and displayStringId properties
  - Added titleproperty to IPropertyItemOptions and IPropertyItem APIs, removed titleAltText and titleStringId properties
  - Added title and tooltip properties to ModalToolCreationParameters API, removed titleAltText, titleStringId, tooltipAltText and tooltipStringId properties

---   

Thank you for being here! You're the best! ❤️

**Comment by AbbottMc** [comment]
Great Update!
